openmetadata-service/src/main/java/org/openmetadata/service/security/auth/LdapAuthenticator.java

```diff
@@ -0,0 +1,250 @@
+import org.openmetadata.service.exception.EntityNotFoundException;
+import org.openmetadata.service.jdbi3.TokenRepository;
+import org.openmetadata.service.jdbi3.UserRepository;
+import org.openmetadata.service.security.AuthenticationException;
+import org.openmetadata.service.util.EmailUtil;
+import org.openmetadata.service.util.TokenUtil;
+
+@Slf4j
+public class LdapAuthenticator implements AuthenticatorHandler {
+  private final UserRepository userRepository;
+  private final TokenRepository tokenRepository;
+  private final LoginAttemptCache loginAttemptCache;
+  private final LdapConfiguration ldapConfiguration;
+  private final LDAPConnectionPool ldapLookupConnectionPool;
+
+  public LdapAuthenticator(
+      UserRepository userRepository, TokenRepository tokenRepository, OpenMetadataApplicationConfig config) {
+    if (config.getAuthenticationConfiguration().getProvider().equals("ldap")
+        && config.getAuthenticationConfiguration().getLdapConfiguration() != null) {
+      ldapLookupConnectionPool = getLdapConnectionPool(config.getAuthenticationConfiguration().getLdapConfiguration());
+    } else {
+      throw new RuntimeException("Invalid or Missing Ldap Configuration.");
+    }
+    this.userRepository = userRepository;
+    this.tokenRepository = tokenRepository;
+    this.ldapConfiguration = config.getAuthenticationConfiguration().getLdapConfiguration();
+    this.loginAttemptCache = new LoginAttemptCache(config);
+  }
+
+  private LDAPConnectionPool getLdapConnectionPool(LdapConfiguration ldapConfiguration) {
+    try {
+      if (ldapConfiguration.getSslEnabled()) {
+        AggregateTrustManager trustManager =
+            new AggregateTrustManager(
+                false,
+                JVMDefaultTrustManager.getInstance(),
+                new TrustStoreTrustManager(
+                    ldapConfiguration.getKeyStorePath(),
+                    ldapConfiguration.getKeyStorePassword().toCharArray(),
+                    ldapConfiguration.getTruststoreFormat(),
+                    true));
+        SSLUtil sslUtil = new SSLUtil(trustManager);
+
+        LDAPConnectionOptions connectionOptions = new LDAPConnectionOptions();
+        SSLSocketVerifier sslSocketVerifier =
+            ldapConfiguration.getVerifyCertificateHostname()
+                ? new HostNameSSLSocketVerifier(true)
+                : TrustAllSSLSocketVerifier.getInstance();
+        connectionOptions.setSSLSocketVerifier(sslSocketVerifier);
+
+        try (LDAPConnection connection =
+            new LDAPConnection(
+                sslUtil.createSSLSocketFactory(),
+                connectionOptions,
+                ldapConfiguration.getHost(),
+                ldapConfiguration.getPort())) {
+          // Use the connection here.
+          return new LDAPConnectionPool(connection, ldapConfiguration.getMaxPoolSize());
+        } catch (GeneralSecurityException e) {
+          LOG.warn("[LDAP] Issue in creating a LookUp Connection SSL");
```

Comment:
should we throw a IllegalStateException , if we only log warning and server starts users might think server is started fine but ldap is not working